dev_scripts/qa.py

```diff
@@ -0,0 +1,448 @@
+#!/usr/bin/env python3
+
+import argparse
+import enum
+import logging
+import os
+import pathlib
+import subprocess
+import sys
+
+logger = logging.getLogger(__name__)
+
+# The titles of the QA scenarios, as taken from:
+# https://github.com/freedomofpress/dangerzone/blob/main/RELEASE.md#scenarios
+QA_SCENARIOS = [
+    "1. Dangerzone correctly identifies that Docker/Podman is not installed",
+    "2. Dangerzone correctly identifies that Docker is not running",
+    "3. Dangerzone successfully installs the container image",
+    "4. Dangerzone retains the settings of previous runs",
+    "5. Dangerzone reports failed conversions",
+    "6. Dangerzone succeeds in converting multiple documents",
+    "7. Dangerzone CLI succeeds in converting multiple documents",
+    "8. Dangerzone can open a document for conversion via right-click -> 'Open With'",
+    "9. Updating Dangerzone handles external state correctly",
+]
+
+
+class Platform(enum.Enum):
+    """List of available platforms for QA."""
+
+    WINDOWS = "windows"
+    UBUNTU_22_04 = "ubuntu-22.04"
+    UBUNTU_22_10 = "ubuntu-22.10"
+    FEDORA_36 = "fedora-36"
+    FEDORA_37 = "fedora-37"
+
+
+class QABase:
+    """Base class for the QA tasks."""
+
+    def __init__(self, try_auto=False, skip_manual=False, debug=False):
+        self.debug = debug
+        self.try_auto = try_auto
+        self.skip_manual = skip_manual
+        self.src = (
+            subprocess.run(
+                [
+                    "git",
+                    "rev-parse",
+                    "--show-toplevel",
+                ],
+                stdout=subprocess.PIPE,
+            )
+            .stdout.decode()
+            .strip("\n")
+        )
+
+    def auto(*msgs, ref=None):
```

Comment:
I had done this in one of the revisions of the script, but I reverted because it seemed like an overkill. You don't gain much because all it does is print the title of the step through the decorator, which you could do already. But now that we are discussing about anchors in the docs, I'll add it.